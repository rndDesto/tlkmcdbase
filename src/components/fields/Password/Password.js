import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Password(props) {
  const { className, input, inputProps, label } = props;
  const classes = [styles.root, className].filter(Boolean).join(' ');
  const [show, setShow] = useState(false);

  return (
    <div className={classes}>
      <label>{label}</label>
      <div className={className}>
        <input id={input.name} type={show ? 'text' : 'password'} {...input} {...inputProps} />
        <button onClick={() => setShow(!show)} type="button">
          <img alt={show ? 'show' : 'hide'} src={`/assets/ic-password-${show ? 'show' : 'hide'}.svg`} />
        </button>
      </div>
    </div>
  );
}

Password.defaultProps = {
  className: '',
  input: {},
  inputProps: {},
  label: ''
};

Password.propTypes = {
  className: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  label: PropTypes.string
};
