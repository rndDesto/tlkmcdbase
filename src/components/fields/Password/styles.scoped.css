.root > div {
  align-items: center;
  background-color: #FFFFFF;
  border: 0.0625rem solid var(--palette-text-light-2);
  border-radius: 0.25rem;
  display: flex;
  margin-top: 0.625rem;
  padding: 0.75rem 1rem 0.75rem 0.5rem;
  transition: border 0.25s;
}

.root > div.focus {
  border-color: var(--palette-base-primary-main);
}

.root > label {
  color: var(--palette-black-20);
  font-size: 0.875rem;
  line-height: 1rem;
  font-weight: bold;
}

.root > div > input {
  border: none;
  box-shadow: none;
  color: var(--palette-black-10);
  display: block;
  flex: 1;
  font-size: 0.875rem;
  font-weight: bold;
  line-height: 1.5rem;
}

.root > div > input::placeholder {
  color: var(--palette-text-light-2);
}

.root > div > button {
  background-color: transparent;
  border: none;
  cursor: pointer;
  height: 1.5rem;
  width: 1.5rem;
}

.root > div > button > img {
  width: 100%;
}
