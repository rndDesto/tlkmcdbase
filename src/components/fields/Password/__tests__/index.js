import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Password from '../Password';

describe('src/components/fields/Password', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Password />);
    expect(tree).toMatchSnapshot();

    tree.props.children[1].props.children[1].props.onClick();
  });
});
