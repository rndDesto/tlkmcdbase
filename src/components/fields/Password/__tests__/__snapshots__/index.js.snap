// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Password render 1`] = `
<div
  className=""
>
  <label>
    
  </label>
  <div
    className=""
  >
    <input
      type="password"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt="hide"
        src="/assets/ic-password-hide.svg"
      />
    </button>
  </div>
</div>
`;
